### ISSUE TITLE
XML Deserialization Error When CharSet Used In Content-Type

### ISSUE THREAD

**Issue by bpruitt-goddard** [ServiceStack/Issues#237]
When sending an XML request to ServiceStack with the Content-Type HTTP header specifying the charset, the request isn't deserialized at all if the content-type string looks like this "Content-Type: text/xml; charset=utf-8". This is a specific scenario as capitalizing utf or removing the space before charset allow the request to be successfully deserialized.

**Successful headers**
Content-Type: text/xml   (no charset)
Content-Type: text/xml; charset=UTF-8 (capital UTF)
Content-Type: text/xml;charset=utf-8 (no space)
**Failure headers**
Content-Type: text/xml; charset=utf-8 (lowercase utf, space)

Steps to reproduce:
* Create a DTO with at least one property. Add DataContract/DataMembers for XML support.

```
 [Route("/test", Verbs = "POST")]
 [DataContract(Namespace="")]
 public class test
 {
    [DataMember(Order=0)]
    public string User { get; set; }
 }
```
* Try to post to a request with the Content-Type set as follows:

```
POST http://localhost:8000/test HTTP/1.1
Host: localhost:8000
Accept: application/json
Content-Type: text/xml; charset=utf-8
Content-Length: 37

<test>
  <User>steve</User>
</test>
```

**Expected**: Service would receive test object with User property equal to "steve".
 **Actual**: test object is null.

**Comment by mythz** [comment]
ok this behavior is because it matches the [SOAP 1.1 Content-Type](http://www.w3.org/TR/2000/NOTE-SOAP-20000508/#_Toc478383490) which takes precedence. 

Can you change this to use `application/xml` instead?

**Comment by mythz** [comment]
This should now work to handle `text/xml; charset=utf-8` for XML HTTP Requests [in this commit](https://github.com/ServiceStack/ServiceStack/commit/9d7e593bb0a6869614f539e647d43b41dbcb0d54).

This change is available from v4.0.37+ that's now [available on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet)
